File: Python/Multiples_of_3_and_5.py
"""
If we list all the natural numbers below 10 that are multiples of 3
or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.
Find the solution so that it returns the sum of all the multiples of
3 or 5 below the number passes in.

>>> solution(10)
23

"""
def solution(number):
    tot, count = 0, 1
    while count < number:
        if count % 3 == 0 or count % 5 == 0:
            tot = tot + count
            count += 1
        elif count % 15 == 0:
            tot = tot + count
            count += 1
        else:
            count += 1
    return tot

""" Another solution """
#def solution(number):
#    return sum(x for x in range(number) if x % 3 == 0k or x % 5 == 0)

File: Java/Multiples_of_3_and_5.java
// Modified: 07/13/2017
//
// Author: jguardado39 (John Guardado)

/*

https://www.codewars.com/kata/514b92a657cdc65150000006

If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.

Finish the solution so that it returns the sum of all the multiples of 3 or 5 below the number passed in.

Note: If the number is a multiple of both 3 and 5, only count it once.

Courtesy of ProjectEuler.net

*/

public class Solution {

  public int solution(int number) {
    int count = 0;
    for (int i = 0; i < number; i++) {
      if (i % 3 == 0 || i % 5 == 0) {
        count += i;
      }
    }
    return count;
  }
}


// By flekken
import java.util.stream.IntStream;

public class Solution {
  public int solution_alt(int number) {
    return IntStream.range(0, number)
                    .filter(n -> (n % 3 == 0) || (n % 5 == 0))
                    .sum();
  }
}